viewer/packages/cad-geometry-loaders/src/CadModelUpdateHandler.ts

```diff
@@ -178,7 +180,8 @@ export class CadModelUpdateHandler {
             array.push(model);
             return array;
           case 'remove':
-            return array.filter(x => x.cadModelMetadata.modelIdentifier !== model.cadModelMetadata.modelIdentifier);
+            array = array.filter(x => x.cadModelMetadata.modelIdentifier !== model.cadModelMetadata.modelIdentifier);
+            return array;
```

Comment:
Because I was thinking that it could be a reason for unwanted reference, as old ray still holds model value inside. But probably it is erased in any case and replaced with new one, right?

Reply:
Also, in case of new model we are modifying passed array, not creating a new one, so we should probably do the same for both.

Reply:
Hmm, I dont think that would happen - the array will go out of scope. Also, you are not modifying the original `array` - only the local reference to it (i.e. you change the reference, not the content).